### PR TITLE
Standardized attention mask field in DPO, RM and finegrained preferences

### PR DESCRIPTION
**Pull request by abaheti95**
Changed all occurrences of `text_attention_mask` to `attention_mask` following the similar standard of llm-foundry

### CODE REVIEW

**Review by bcui-db — APPROVED** [review]
LGTM awesome thank you!
